### PR TITLE
Create IsItUp helm chart for public use #94

### PR DESCRIPTION
**Pull request by alimer08**
Hello Team!

I have created IsItUp Helm Chart and it has been tested by me and it is working.
Please review, test, and approve it.

Thank you!
Ali M

### CODE REVIEW

**Review by osmdilya — APPROVED** [review]

**Review by omiradil — APPROVED** [review]

File: stable/isitup/JenkinsBuilder.groovy [hunk 1]
```diff
@@ -0,0 +1,3 @@
+@Library('CommonLib@master') _
+def common = new com.lib.JenkinsCommonDockerBuildPipeline()
+common.runPipeline()
```
Comment (fsadykov) [inline stable/isitup/JenkinsBuilder.groovy:3]: Why we have groovy files here?

File: stable/isitup/deployment/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+## Copy everything to image folder root
+WORKDIR /app
+COPY ./isitup /app
+COPY ./isitup/configurations/domains.yaml /app/domains.yaml
+
+## Putting inside Docker file which environment is 
+ARG version
+ENV RELEASE=$version
+
+## Create kube folder insied root Copy KuberConfig to the docker images
```
Comment (fsadykov) [inline stable/isitup/deployment/docker/Dockerfile:12]: Why we have docker files and the path of the files are incorrect

File: stable/isitup/deployment/docker/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+drwxrwxr-x. 3 vrodi18 vrodi18  82 May 12 16:19 .
+drwxrwxr-x. 4 vrodi18 vrodi18  37 May  7 04:05 ..
+-rw-rw-r--. 1 vrodi18 vrodi18 885 May 10 00:24 docker-compose.yaml
+-rw-rw-r--. 1 vrodi18 vrodi18 549 May  7 04:05 Dockerfile
+drwxrwxr-x. 5 vrodi18 vrodi18 125 May  7 04:05 isitup
+-rw-rw-r--. 1 vrodi18 vrodi18   0 May 12 16:19 README.md
+```
+
+
+Docker compose has environment variables configured see below all env
```
Comment (fsadykov) [inline stable/isitup/deployment/docker/README.md:33]: Why we need the documentation about docker compose in helm chart strucure?

File: stable/isitup/deployment/docker/docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+## You will need to export following credentials to run the app <GIT_TOKEN> <GITHUB_CLIENT_ID> <GITHUB_CLIENT_SECRET>
+  environment: 
+   GIT_TOKEN: ${GIT_TOKEN}
+  #  GITHUB_CLIENT_ID: ${GITHUB_CLIENT_ID}
+  #  GITHUB_CLIENT_SECRET: ${GITHUB_CLIENT_SECRET}
+   GOOGLE_APPLICATION_CREDENTIALS: /root/.kube/flask-kube.json
+  entrypoint: ["/bin/sh","-c"]
+
+
```
Comment (fsadykov) [inline stable/isitup/deployment/docker/docker-compose.yaml:20]: We don't need docker compose files

File: stable/isitup/deployment/docker/isitup/app.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+@app.route('/re-sync')
+def re_sync():
+    with open('configurations/user-information.yaml', 'w') as file:
+        config_data = {
+            "data" : []
+        } 
+        for user in org.get_members():
+            config_data['data'].append(user.raw_data)
+        yaml.dump(config_data, file)
+    return jsonify({"message": "Successful"})
```
Comment (fsadykov) [inline stable/isitup/deployment/docker/isitup/app.py:44]: We don't need the application in helm chart structure

File: stable/isitup/deployment/terraform/charts/isitup/values.yaml [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+  - name: MYSQL_DATABASE
+    valueFrom:
+      secretKeyRef:
+        name: ${isitup_credentials}
+        key: mysql_database
+      
+  - name: MYSQL_PASSWORD
+    valueFrom:
+      secretKeyRef:
+        name: ${isitup_credentials}
```
Comment (fsadykov) [inline stable/isitup/deployment/terraform/charts/isitup/values.yaml:56]: The helm chart still using terraform variables

File: stable/isitup/deployment/terraform/output.tf [hunk 7]
```diff
@@ -0,0 +1,13 @@
+data "template_file" "output" {
+  template = "${file("output.txt")}"
+
+  vars {
+    Chart_version  = "${var.deployment_name}-${var.release_version}"
+    Helm_release_name = "${var.deployment_name}-${var.deployment_environment}"
+    Domain_name  = "${lookup(var.deployment_endpoint, "${var.deployment_environment}")}.${var.google_domain_name}"
+  }
```
Comment (fsadykov) [inline stable/isitup/deployment/terraform/output.tf:8]: Delete the terraform files

**Review by fsadykov — CHANGES_REQUESTED** [review]
This PR definitely needs to work on!!

File: stable/isitup-test/.helmignore [hunk 8]
```diff
@@ -0,0 +1,23 @@
+# Patterns to ignore when building packages.
+# This supports shell glob matching, relative path matching, and
+# negation (prefixed with !). Only one pattern per line.
+.DS_Store
+# Common VCS dirs
+.git/
+.gitignore
+.bzr/
```
Comment (fsadykov) [inline stable/isitup-test/.helmignore:8]: Remove isitup-test we don't need this at all

File: stable/isitup/templates/proxy-deploy.yaml [hunk 9]
```diff
@@ -0,0 +1,50 @@
+apiVersion: apps/v1
```
Comment (fsadykov) [inline stable/isitup/templates/proxy-deploy.yaml:1]: Add the if statement for proxy to be enabled or disabled

File: stable/isitup/templates/proxy-ingress.yaml [hunk 10]
```diff
@@ -0,0 +1,41 @@
+apiVersion: networking.k8s.io/v1beta1
```
Comment (fsadykov) [inline stable/isitup/templates/proxy-ingress.yaml:1]: If statement here as well

**Review by fsadykov — CHANGES_REQUESTED** [review]
Needs to work on

**Review by bermet-s — APPROVED** [review]

**Review by rahymov — APPROVED** [review]

**Review by fuchicorp-manage — APPROVED** [review]

**Review by fuchicorp-manage — APPROVED** [review]

**Review by fsadykov — APPROVED** [review]

**Pull request merged**